### PR TITLE
feat(audio-scenarios): Black Flag Furled — 1 s debounce on raised + "furled cleared" callout (#669)

### LINKED ISSUE #669: feat(audio-scenarios): Black Flag Furled — 1s debounce on raised + new "furled cleared" callout
### What's wrong / what we want

The Race Engineer announces the **Black Flag Furled** warning immediately when iRacing raises the `Furled` session-flag bit. Running briefly off track flashes that bit for ~0.5 s even though it isn't a real furled-black-flag warning, so the engineer gives the furled readout when it shouldn't. Two changes, together:

1. **Debounce the raised callout by 1 second.** Only announce once the `Furled` bit has stayed high for ≥ 1 s; if it clears within that second, drop the announcement (nothing plays).
2. **Add a paired "furled cleared" callout** that fires when a *genuine* (announced) furled flag is withdrawn — mirroring the existing `flag.yellow.cleared`. It is gated on the raised callout having fired, so a transient flicker fires **neither** raised nor cleared.

### Steps to reproduce (the bug)

1. In a race, drive off track briefly so iRacing flashes the furled bit.
2. Today the engineer plays the furled callout despite no genuine warning.

### Root cause

`flag.furled.raised` is a plain rising edge in `packages/sim-events-iracing/src/diff/flags.ts` (`case "furled"` in the "new raised transitions" switch), with no time gate. `diffFlags(...)` doesn't receive `now` today (unlike `diffDamage` / `diffLimiter` / `diffToggles`). The only paired "cleared" event today is `flag.yellow.cleared` (falling-edge of `lastAnyYellow`).

### Fix

Reuse two existing patterns, scoped to the furled bit only:

- **Debounce** like `diff/damage.ts` (`DAMAGE_DEBOUNCE_MS` / `damagePendingAt`): thread the per-tick `now` into `diffFlags`; add `FURLED_DEBOUNCE_MS = 1000` (exported for tests); add `furledPendingAt` + `furledAnnounced` to `TranslatorState` (type **and** `createInitialState`). On a genuine `Furled` rising edge start the timer (don't emit); emit `flag.furled.raised` only once the bit has stayed set ≥ 1 s (set `furledAnnounced`); reset the timer the moment the bit clears.
- **Cleared callout** like `flag.yellow.cleared`: on the falling edge, if `furledAnnounced` was true, emit a new `flag.furled.cleared` (`EmptySimEventPayload`); a never-announced flicker emits nothing. New `pit-crew.flag-furled-cleared` scenario via the existing `flagScenario(...)` builder, new `flag-furled-cleared` pool + clip, and a new per-flag opt-in `calloutEnabledFlagFurledCleared` (its own `FlagCalloutId` member, mirroring `yellow-cleared`).

Diff-side, **not** a scenario `triggerDelay` — a deferred scenario doesn't re-run its `where:`, so only the per-tick diff can re-check that the live bit is still set when the window elapses. Furled-only; other flags are sustained and keep their immediate emit. No interpreter / in-flight-cancel changes.

Proposed cleared line (finalize at implementation): "Furled flag's been lifted. You're in the clear."

### Affected artifacts

**Debounce (sim-events-iracing):**
- `packages/sim-events-iracing/src/diff/flags.ts` — debounce the furled raised emit; emit `flag.furled.cleared` on the announced falling edge; accept `now`; export `FURLED_DEBOUNCE_MS`.
- `packages/sim-events-iracing/src/state.ts` — `furledPendingAt` + `furledAnnounced` (type **and** `createInitialState()`).
- `packages/sim-events-iracing/src/translator.ts` — pass `now` to `diffFlags`.

**New `flag.furled.cleared` event + callout:**
- `packages/event-bus/src/event-catalog.ts` — `flag.furled.cleared` (`EmptySimEventPayload`, beside `flag.yellow.cleared`).
- `packages/audio-scenarios/src/catalog/pit-crew/flag-alerts.ts` — new `FURLED_CLEARED` scenario in `FLAG_ALERTS`.
- `packages/audio-scenarios/src/catalog/pit-crew/pools.ts` — `flag-furled-cleared` pool.
- `packages/audio-scenarios/src/catalog/pit-crew/index.ts` — add `"furled-cleared"` to `FlagCalloutId`, `FLAG_CALLOUT_SETTING_KEYS`, `SCENARIO_ID_TO_FLAG_ID`.
- `packages/deck-core/src/global-settings.ts` — `calloutEnabledFlagFurledCleared` Zod field (default `true`).
- `packages/iracing-actions/src/actions/pit-crew/pit-crew.ejs` — add the "Furled cleared" checkbox to the `flagCallouts` list.
- `packages/scenario-harness/src/event-names.ts` — `flag.furled.cleared` template (compile-time completeness check).
- `packages/scenario-harness/src/scenario-shortcuts.ts` — "Furled Cleared" shortcut.
- `packages/audio-assets/configs/default.voice.json` — `furled-cleared-01` entry in the `flags` group; then `generate --group flags` + `generate:manifest`.
- **Plugins:** no change — `getFlagCalloutEnabled` is generic over `FlagCalloutId`.

**Tests / fixtures:**
- `packages/sim-events-iracing/src/diff/flags.test.ts` (+ any `translator.test.ts` call site) — update the `diffFlags` call for `now`; add debounce + cleared tests driving the diff with **explicit `now`** across the 1 s boundary (`translator.test.ts` reads the real `Date.now()` and can't simulate time — test the diff directly, as `diff/pit-lane.test.ts` does for the pit-approach cooldown).
- `packages/audio-scenarios/src/catalog/pit-crew/flag-alerts.test.ts` — add `furled-cleared` to the `FLAG_SCENARIO_IDS` / pool assertions.
- `packages/audio-scenarios/src/catalog/pit-crew/register-pit-crew.test.ts` — add `furled-cleared` to `FLAG_CLIP_NAMES`, `ALL_FLAG_IDS`, `FLAG_FIRES`.
- `packages/deck-core/src/simhub-service.test.ts` — add `calloutEnabledFlagFurledCleared` to **both** exhaustive literals.

**Docs:** `.claude/rules/race-engineer-callout-examples.md` — one entry for the transient-flag-bit debounce + the announced-only paired-cleared pattern.

### Acceptance criteria

- [ ] An off-track furled flicker (< 1 s) produces **no** callout (neither raised nor cleared).
- [ ] A furled flag sustained ≥ 1 s announces the raised callout **once**.
- [ ] When that announced furled is withdrawn, the **furled-cleared** callout fires once.
- [ ] Furled clearing during the 1 s wait cancels the pending raised call and fires no cleared call.
- [ ] Furled clear → re-raise can announce again (fresh episode).
- [ ] Connecting while furled is already up does not blurt on the first tick.
- [ ] Both per-flag opt-ins (`calloutEnabledFlagFurled`, `calloutEnabledFlagFurledCleared`) gate their callout live.
- [ ] `pnpm install`, `pnpm build`, `pnpm test`, `pnpm lint:fix`, `pnpm format:fix` all pass.

### PR DESCRIPTION
**Pull request by niklam**
## Related Issue

Fixes #669

## What changed?

- **`flag.furled.raised` is debounced by 1 s** (`FURLED_DEBOUNCE_MS`, diff-side in `sim-events-iracing/src/diff/flags.ts`): the `Furled` bit's rising edge only arms the window; a brief off-track excursion that flashes the bit for ~0.5 s no longer triggers the callout. Connecting mid-furled stays silent like every other flag.
- **New paired `flag.furled.cleared` event + "furled cleared" callout** ("Black flag cleared."), emitted on the falling edge only when the raised callout was actually announced (`furledAnnounced`) — a transient flicker fires neither event, and clear → re-raise announces again as a fresh episode.
- **Audio wiring**: `pit-crew.flag-furled-cleared` scenario (queueable like `YELLOW_CLEARED`), `flag-furled-cleared` pool, new `FlagCalloutId` member with `calloutEnabledFlagFurledCleared` opt-in (default ON), PI checkbox, scenario-harness event template + "Furled Cleared" shortcut. No `registerPitCrew` or plugin changes — the flag closure is generic over `FlagCalloutId`.
- **Queued-call staleness fix** (manual-test follow-up): the queueable furled fire can defer behind a longer call (e.g. incident points) and deferred fires replay without re-running `where:`. The raised scenario now wraps its whole sequence in an `if:` step that re-checks the live `Furled` bit at speak time — a warning withdrawn while the call sat in the queue expands to silence (radio frame included). The cleared scenario's `where:` is gated on the raised line having actually reached the speaker, so a stray "Black flag cleared." can't follow a never-spoken raise.
- **Voice clip**: `voice/default/flags/furled-cleared-01.mp3` generated via ElevenLabs (`"seed": 1`), both manifests updated.
- **Docs**: website Pit Crew page (Furled bullets + 22 flag callouts), `race-engineer-callout-examples.md` entry (three gates at three layers), audio-assets generation guidance (existing-group walkthrough, dry-run preview, `"seed": 1` default), audio-scenarios CLAUDE.md (existing-family checklist + speak-time `if:`-wrap pattern), stale `generate.config.json` reference fixed in root CLAUDE.md.

## How to test

1. Scenario harness: fire **Furled** → raised line plays; fire **Furled Cleared** → "Black flag cleared." plays (only after a raised line actually played).
2. In iRacing: drive briefly off track so the furled bit flickers (< 1 s) → no callout, and no cleared callout when it drops.
3. Earn a genuine furled black flag (sustained ≥ 1 s) → "We got the furled black flag…" plays once; give the time back so the flag is withdrawn → "Black flag cleared." plays once.
4. Queued case: with other radio traffic playing when the furled bit rises, let the warning clear before the engineer gets the bus → neither the raised nor the cleared line plays.
5. Toggle "Black flag furled" / "Furled cleared" in the Pit Crew PI mid-session → each gate takes effect on the next event without cutting an in-flight clip.

## Checklist

- [x] Linked to an approved issue
- [x] New code has unit tests
- [x] Changes registered in all applicable plugins (Stream Deck, Mirabox) — no plugin-code changes required; the existing `getFlagCalloutEnabled` closure is generic over `FlagCalloutId`
- [x] No unrelated changes included

## Summary by CodeRabbit

* **New Features**
  * Added “Furled cleared” (Black flag cleared) callout; paired with the existing Furled announcement so it only plays if the raised callout actually ran.
  * New settings toggle to enable/disable the Furled cleared callout.

* **Bug Fixes / Reliability**
  * Furled announcements are debounced to avoid spurious alerts from brief flag flickers; cleared pairing and queued replay remain consistent.

* **Documentation**
  * Expanded audio generation and authoring guides with dry-run, scoped generation, manifest rebuild, and pool selection guidance.

### CODE REVIEW

File: packages/audio-scenarios/src/catalog/pit-crew/flag-alerts.ts [hunk 1]
```diff
@@ -219,15 +219,92 @@ const DISQUALIFY: Scenario = {
...
+  when: {
+    event: "flag.furled.cleared",
+    where: () => {
+      if (!furledRaisedSpoken) return false;
+
+      furledRaisedSpoken = false;
+
+      return true;
+    },
+  },
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/audio-scenarios/src/catalog/pit-crew/flag-alerts.ts]: Good catch — accepted, with one deliberate deviation from the suggested diff. Fixed in e6f9c7ce.

The clear now mirrors the raised scenario: `where:` is a passive read of `furledRaisedSpoken`, and a speak-time `if:` wraps the whole sequence, re-checks the live bit, and consumes the marker only when the line actually plays — so a queued clear dies silently when the warning is back up, and the kept marker still pairs the eventual genuine drop.

The deviation: the suggestion reuses `furledStillUp()`, whose missing-telemetry fallback is "up" (correct for the raised gate per the don't-punish-missing-data precedent, #574). Reused on the clear gate, that fallback would permanently suppress "Black flag cleared." in the scenario harness and on disconnect. The helper is now `furledBitUp(fallbackWhenUnknown)` with inverted fallbacks per gate: raised treats unknown as still-up, clear treats unknown as not-re-raised — each side passes the value that keeps it from suppressing on missing data.

Covered by two new tests: the immediate bit-up-again case (marker kept for the real drop) and the queued end-to-end case (filler occupying the bus, clear queued, bit back up at replay → silence, then the genuine drop still pairs).

**Comment by niklam** [comment]
Re the outside-diff comment ("Use \"random no-repeat\" wording for multi-element pools"): valid — `pickFromPool` samples uniform-random with a no-immediate-repeat guard, and #657 already corrected `pools.ts`/`flag-alerts.ts` but missed the guidance docs. Fixed in 1fda7b06 across all three stale spots: `.claude/rules/race-engineer-callouts.md` (step 4), `packages/audio-assets/CLAUDE.md` (variants convention), and `packages/audio-scenarios/CLAUDE.md` (Pools section, which also said "rotate").
